src/Engine/ProtoCore/Utils/CoreUtils.cs

```diff
@@ -825,5 +825,33 @@ public static CodeBlock GetCodeBlock(List<CodeBlock> blockList, int blockId)
             }
             return codeblock;
         }
+
+        public static ProcedureNode GetFirstVisibleProcedure(string name, List<Type> argTypeList, CodeBlock codeblock)
+        {
+            Validity.Assert(null != codeblock);
+            if (null == codeblock)
```

Comment:
no need to add null check here. `Validity.Assert()` throws an exception if the condition fail, so never get a chance to continue.